k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java

```diff
@@ -83,7 +83,16 @@ public final void parse(InputStream in) throws IOException, MessagingException {
         parse(in, false);
     }
 
-    private void parse(InputStream in, boolean recurse) throws IOException, MessagingException {
+    public final void parse(InputStream in, boolean recurse) throws IOException, MessagingException {
+        MimeConfig parserConfig  = new MimeConfig();
+        parserConfig.setMaxHeaderLen(-1); // The default is a mere 10k
+        parserConfig.setMaxLineLen(-1); // The default is 1000 characters. Some MUAs generate
+        // REALLY long References: headers
+        parserConfig.setMaxHeaderCount(-1); // Disable the check for header count.
+        parse(in, recurse, parserConfig);
+    }
+
+    public void parse(InputStream in, boolean recurse, MimeConfig parserConfig) throws IOException, MessagingException {
```

Comment:
If we move `MessageTest` to the internet package, these can be package-local. Also please annotate `@VisibleForTesting`